dcos/util.py

```diff
@@ -130,6 +130,7 @@ def ensure_file_exists(path):
 
     if not os.path.exists(path):
         try:
+            ensure_dir_exists(os.path.dirname(path))
```

Comment:
We should only create directory for DCOS_CONFIG when the value is the default. We should not create the directories when the user has overwritten their DCOS_CONFIG.